pretty_midi/instrument.py

```diff
@@ -113,6 +125,25 @@ def get_piano_roll(self, fs=100, times=None):
             piano_roll[note.pitch,
                        int(note.start*fs):int(note.end*fs)] += note.velocity
 
+        # Process sustain pedals
+        if use_pedal:
+            CC_SUSTAIN_PEDAL = 64
+            time_pedal_on = 0
+            is_pedal_on = False
+            for cc in [_e for _e in self.control_changes
+                       if _e.number == CC_SUSTAIN_PEDAL]:
+                time_now = int(cc.time*fs)
+                is_current_pedal_on = (cc.value > pedal_threshold)
+                if not is_pedal_on and is_current_pedal_on:
+                    time_pedal_on = time_now
+                    is_pedal_on = True
+                elif is_pedal_on and not is_current_pedal_on:
+                    subpr = piano_roll[:, time_pedal_on:time_now]
+                    pedaled = np.minimum(subpr.cumsum(1),
+                                         subpr.max(1)[:, np.newaxis])
```

Comment:
I'm a little confused by this operation.  Isn't `subpr.cumsum(1)` always less than `subpr.max(1)` given that subpr is positive (and it better be)? If you want to dress all of this code up with comments it wouldn't hurt!